src/renderer/components/+cluster-settings/components/show-metrics.tsx

```diff
@@ -0,0 +1,58 @@
+import "./cluster-metrics-setting.scss";
+
+import React from "react";
+import { observer } from "mobx-react";
+import { Cluster } from "../../../../main/cluster";
+import { observable } from "mobx";
+import { Badge } from "../../badge/badge";
+import { Icon } from "../../icon/icon";
+
+interface Props {
+  cluster: Cluster;
+}
+
+@observer
+export class ShowMetricsSetting extends React.Component<Props> {
+  @observable hiddenMetrics = observable.set<string>();
+
+  componentDidMount() {
+    this.hiddenMetrics = observable.set<string>(this.props.cluster.preferences.hiddenMetrics);
+  }
+
+  removeMetric = (metric: string) => {
+    this.hiddenMetrics.delete(metric);
+    this.props.cluster.preferences.hiddenMetrics = Array.from(this.hiddenMetrics);
+  };
+
+  renderMetrics() {
+    const hiddenMetrics = this.props.cluster.preferences.hiddenMetrics;
+
+    return (
+
+      Array.from(hiddenMetrics).map(name => {
+        const tooltipId = `${name}`;
+
+        return (
+          <Badge key={name}>
+            <span id={tooltipId}>{name}</span>
+            <Icon
+              small
+              material="clear"
+              onClick={() => this.removeMetric(name)}
+              tooltip="Remove"
+            />
+          </Badge>
+        );
+      })
+    );
```

Comment:
For rendering we should be using the set. 

```suggestion
    return Array.from(this.hiddenMetrics)
      .map(name => (
        <Badge key={name}>
          <span id={name}>{name}</span>
          <Icon
            small
            material="clear"
            onClick={() => this.removeMetric(name)}
            tooltip="Remove"
          />
        </Badge>
      ));
```